pandas/_libs/tslibs/conversion.pyx

```diff
@@ -849,16 +853,21 @@ def tz_localize_to_utc(ndarray[int64_t] vals, object tz, object ambiguous=None,
         ndarray ambiguous_array
         Py_ssize_t i, idx, pos, ntrans, n = len(vals)
         int64_t *tdata
-        int64_t v, left, right
+        int64_t v, left, right, val, v_left, v_right
         ndarray[int64_t] result, result_a, result_b, dst_hours
         npy_datetimestruct dts
         bint infer_dst = False, is_dst = False, fill = False
+        bint shift = False, fill_nonexist = False
         bint is_coerce = errors == 'coerce', is_raise = errors == 'raise'
 
     # Vectorized version of DstTzInfo.localize
-
     assert is_coerce or is_raise
 
+    if is_coerce:
```

Comment:
can we do the warning at a higher level? eg. not in the cython code? IOW refactor to avoid passing errors at all